### PR TITLE
Hw 35

### PR DESCRIPTION
**Pull request by akutil**

### CODE REVIEW

File: src/main/java/com/gmail/kutilandrej/cinema/controller/InjectDataController.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+public class InjectDataController {
+
+    @Autowired
+    private RoleService roleService;
+    @Autowired
+    private UserService userService;
+
+    @PostConstruct
+    public void insertRolesToDb() {
+        Role r1 = roleService.add(Role.of("ADMIN"));
```
Comment (Sofasmile) [inline src/main/java/com/gmail/kutilandrej/cinema/controller/InjectDataController.java]:
```suggestion
        Role roleAdmin = roleService.add(Role.of("ADMIN"));
```

File: src/main/java/com/gmail/kutilandrej/cinema/controller/InjectDataController.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        Role r2 = roleService.add(Role.of("USER"));
+        User user2 = new User("admin", "admin@i.ua", "1234", Set.of(r1));
```
Comment (Sofasmile) [inline src/main/java/com/gmail/kutilandrej/cinema/controller/InjectDataController.java]:
```suggestion
        User admin = new User("admin", "admin@i.ua", "1234", Set.of(r1));
```

File: src/main/java/com/gmail/kutilandrej/cinema/dao/impl/RoleDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+    @Override
+    public Role getRoleByName(String roleName) {
+        try (Session session = sessionFactory.openSession()) {
+            String hql = "FROM Role R WHERE R.roleName = :roleName";
+            Query<Role> query = session.createQuery(hql, Role.class);
+            query.setParameter("roleName", roleName);
+            return query.uniqueResult();
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't find User", e);
```
Comment (Sofasmile) [inline src/main/java/com/gmail/kutilandrej/cinema/dao/impl/RoleDaoImpl.java]: message is wrong

File: src/main/java/com/gmail/kutilandrej/cinema/security/CustomUserDetailsService.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CustomUserDetailsService implements UserDetailsService {
+
+    @Autowired
+    UserService userService;
```
Comment (Sofasmile) [inline src/main/java/com/gmail/kutilandrej/cinema/security/CustomUserDetailsService.java]: access modifier

File: src/main/java/com/gmail/kutilandrej/cinema/security/CustomUserDetailsService.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+    @Override
+    public UserDetails loadUserByUsername(String s) throws UsernameNotFoundException {
+        Optional<User> user = userService.findByEmail(s);
+        UserBuilder builder;
+        if (user.isPresent()) {
+            builder = org.springframework.security.core.userdetails.User.withUsername(s);
+            builder.password(user.get().getPassword());
+            String[] roles = user.get().getRoles().stream().map(Role::getRoleName)
+                    .map(Enum::name)
```
Comment (Sofasmile) [inline src/main/java/com/gmail/kutilandrej/cinema/security/CustomUserDetailsService.java]: can we use only one `map` method?

File: src/main/java/com/gmail/kutilandrej/cinema/service/impl/AuthenticationServiceImpl.java [hunk 6]
```diff
@@ -16,21 +15,18 @@
...
     @Autowired
     private ShoppingCartService shoppingCartService;
-    @Autowired
-    private HashUtil hashUtil;
 
     @Override
     public User login(String email, String password) throws AuthenticationException {
         return userService.findByEmail(email).filter(u -> u.getPassword()
-                .equals(hashUtil.hashPassword(password, u.getSalt())))
+                .equals(password))
```
Comment (Sofasmile) [inline src/main/java/com/gmail/kutilandrej/cinema/service/impl/AuthenticationServiceImpl.java]: https://stackoverflow.com/a/26818730

File: src/main/java/com/gmail/kutilandrej/cinema/security/CustomUserDetailsService.java (hunk 5 above)
Comment (akutil) [reply src/main/java/com/gmail/kutilandrej/cinema/security/CustomUserDetailsService.java]: Yes, we can.

File: src/main/java/com/gmail/kutilandrej/cinema/service/impl/AuthenticationServiceImpl.java (hunk 6 above)
Comment (akutil) [reply src/main/java/com/gmail/kutilandrej/cinema/service/impl/AuthenticationServiceImpl.java]: I deleted this method)

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**